test/e2e/fqdn_dns_cache_test.go

```diff
@@ -0,0 +1,459 @@
+package e2e
+
+import (
+	crdv1beta1 "antrea.io/antrea/pkg/apis/crd/v1beta1"
+	agentconfig "antrea.io/antrea/pkg/config/agent"
+	"antrea.io/antrea/test/e2e/utils"
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/stretchr/testify/require"
+	"gopkg.in/yaml.v2"
+	v1 "k8s.io/api/core/v1"
+	v12 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"strings"
+	"testing"
+	"time"
+)
+
+/*
+1) Create the KIND cluster.
+2) Once cluster is up , create a  service.
+3) Get the IP of above service and configure the same in antrea configMap.
+4) Update antrea configMap.
+
+5) Create NGINX deployment.
+6) Get IP of one of the pods of nginx.
+7) create and configure the custom CoreDNS configMap with the IP received above.
+8) Create custom CoreDNS deployment.
+9) Create and apply antrea FQDN policy.
+10) Deploy antrea-toolbox.
+
+
+---------- tic
+11) curl the FQDN from within toolbox.
+12) imitate caching the IP belonging to above FQDN resolution by keeping it in a variable.
+13) edit configmap with the other IP.
+14) wait for new IP to get updated in configMap and let the changes be reflected in dns pod.
+15) curl the FQDN again with IP , simulating usage of cache -- and it must fail with no connectivity.
+*/
+
+func TestFQDNPolicyWithCachedDNS(t *testing.T) {
```

Comment:
This function is _way_ too long and is not very readable. Please create smaller functions (e.g. for step 1-4, call it `configureAntreaForCachedDNSTests`)